third_party/meerkat/Component/mmSH/android/java/src/com/samsung/android/meerkat/MeerkatServerService.java

```diff
@@ -89,7 +93,19 @@ public int onStartCommand(Intent intent, int flags, int startId) {
         sharedPreferences = applicationContext.getSharedPreferences(
                 "com.samsung.android.meerkat.CAPABILITY", Context.MODE_PRIVATE);
                 sharedPreferences.registerOnSharedPreferenceChangeListener(this);
-        setCapability(sharedPreferences.getString("capability", ""));
+
+        JSONParser parser = new JSONParser();
+        JSONObject newCapability = new JSONObject();
+        for (Map.Entry<String, ?> capability : sharedPreferences.getAll().entrySet()) {
+            try {
+                Log.d("Capability [", capability.getKey().toString() + "] : " + capability.getValue().toString());
+                JSONObject jsonObj = (JSONObject) parser.parse(capability.getValue().toString());
+                newCapability.put(capability.getKey().toString(), jsonObj);
+            } catch (ParseException e) {
+                Log.e(TAG, "JSON parse error : " + e);
+            }
+        }
```

Comment:
I think this is a part of initialization of capabilities, how about moving it into `initCapability()`?

Reply:
Done.